src/components/interface/capmgr/stubs/c_stub.c

```diff
@@ -0,0 +1,203 @@
+#include <memmgr.h>
+#include <capmgr.h>
+#include <cos_thd_init.h>
+#include <cos_defkernel_api.h>
+
+u32_t capmgr_initthd_create_intern(spdid_t c, spdid_t s);
+u32_t capmgr_thd_create_intern(spdid_t c, int idx);
+u32_t capmgr_ext_thd_create_intern(spdid_t c, spdid_t s, int idx);
+u32_t capmgr_aep_create_intern(arcvcap_t *rcvret, tcap_t *tcret, spdid_t c, int idx, int owntc);
+u32_t capmgr_ext_aep_create_intern(arcvcap_t *rcvret, u32_t *rcvtcret, spdid_t c, spdid_t s, int idx, int owntc);
+u32_t capmgr_initaep_create_intern(asndcap_t *sndret, u32_t *rcvtcret, spdid_t c, spdid_t s, int owntc);
+u32_t capmgr_thd_retrieve_intern(spdid_t c, spdid_t s, thdid_t t);
+u32_t capmgr_thd_retrieve_next_intern(spdid_t c, spdid_t s);
+asndcap_t capmgr_asnd_create_intern(spdid_t c, spdid_t s, thdid_t t);
+
+#define __THDID_THDCAP(val, tid, thd) do { (tid) = ((val) >> 16); (thd) = ((val) << 16) >> 16; } while (0);
+
+thdcap_t
+capmgr_thd_retrieve_next(spdid_t child, thdid_t *tid)
+{
+	u32_t ret;
+	thdcap_t thd;
+
+	ret = capmgr_thd_retrieve_next_intern(0, child);
```

Comment:
wink.. I'll change it. 
I think it was when I had to have `unused` variables (not supporting variable args) and I wanted to avoid it as much as possible.